events/syscall/unexpected_k8s_nodeport_connection.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syscall
+
+import (
+	"github.com/falcosecurity/event-generator/events"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+var _ = events.Register(
+	UnexpectedK8sNodePortConnection,
+	events.WithDisabled(), // this rule is not included in falco_rules.yaml (stable rules), so disable the action
+)
+
+func UnexpectedK8sNodePortConnection(h events.Helper) error {
+	if h.InContainer() {
+		port := 31000
+
+		// Get the host's eth0 IP address
+		cmd := exec.Command("ip", "route", "show", "default")
+		hostEth0IP, err := cmd.Output()
```

Comment:
Question: are we sure there's no way to obtain this just using Go code (ie. without using an external util, like `ip`)?
If yes, it would be preferable. 
If not, ok to use `ip` (unless @FedeDP has a better idea :innocent: )

Reply:
@leogr I made this change,
I've added a new function getHostEth0IP()
that uses the net package to find the host's eth0 IP address. iterates through the network interfaces, finds the "eth0" interface, and then iterates through its associated IP addresses to find the first non-loopback IPv4 address.

Reply:
@leogr I made this change,
I've added a new function `getHostEth0IP()`, It

- uses the net package to find the host's eth0 IP address. 
- iterates through the network interfaces
- finds the "eth0" interface
- then iterates through its associated IP addresses to find the first non-loopback IPv4 address.

Reply:
I tested this event inside container, and it is working fine and triggering rule 
![Screenshot from 2024-04-12 16-09-45](https://github.com/falcosecurity/event-generator/assets/117383853/4fe19674-bb7b-4518-8e9f-e8d6ee62c281)